api/tests/devcomp/unit/infrastructure/datasources/learning-content/validation/utils.js

```diff
@@ -1,7 +1,37 @@
 import Joi from 'joi';
+import { HtmlValidate } from 'html-validate';
 
 const uuidSchema = Joi.string().guid({ version: 'uuidv4' }).required();
 
 const proposalIdSchema = Joi.string().regex(/^\d+$/);
 
-export { uuidSchema, proposalIdSchema };
+const htmlValidate = new HtmlValidate();
+const severity = ['', 'Warning', 'Error'];
+const htmlSchema = Joi.string()
+  .external(async (value) => {
+    const report = await htmlValidate.validateString(value);
+
+    if (!report.valid) {
+      let errorLog = '';
+      errorLog = errorLog.concat('\n', `${report.errorCount} error(s), ${report.warningCount} warning(s)\n`);
+      errorLog = errorLog.concat('\n', '─'.repeat(60));
+      for (const result of report.results) {
+        const line = result.source ?? '';
+        for (const message of result.messages) {
+          errorLog = errorLog.concat('\n');
+          errorLog = errorLog.concat('\n', severity[message.severity], `(${message.ruleId}): `, message.message);
+          errorLog = errorLog.concat('\n', message.ruleUrl);
+          errorLog = errorLog.concat('\n');
+          errorLog = errorLog.concat('\n', line);
+          errorLog = errorLog.concat('\n');
+          errorLog = errorLog.concat('\n', '─'.repeat(60));
+          errorLog = errorLog.concat('\n');
+        }
+      }
```

Comment:
Reprise de https://html-validate.org/guide/api/getting-started.html#displaying-the-results

Reply:
Ce parsing d'erreur m'a poussé à créer le ticket `PIX-11317` pour que les erreurs JOI soient + claires pour la contrib.

Reply:
J'avais déjà initié https://1024pix.atlassian.net/browse/PIX-11303 ;)